tests/5.0/declare_target/test_declare_target_device_type_nohost.c

```diff
@@ -7,49 +7,62 @@
 // clause is present on the contained declare target directive, then its argument 
 // determines which versions are made available. If device_type(nohost) is present 
 // only a device version of the procedure is made available. 
+// The device version of the function is specified via declare variant.
 // 
 //===-------------------------------------------------------------------------------===//
 
 #include <omp.h>
 #include <stdio.h>
+#include <assert.h>
 #include <stdlib.h>
 #include "ompvv.h"
 
-#define N 1024
+#define N 10
 int errors = 0;
 
+void target_function();
+
 #pragma omp declare target
 int a[N], b[N], c[N];  
 int i = 0;
 #pragma omp end declare target
 
-void update() { 
+//#pragma omp declare variant(target_function) match(construct={target})
+#pragma omp declare variant(target_function) match(device={kind(nohost)})
+void update() {
+  for (i = 0; i < N; i++) {
+    a[i] += 5;
+    b[i] += 5;
+    c[i] += 5;
+  }
+}
+//#pragma omp end declare variant
+
+#pragma omp declare target device_type(nohost)
+void target_function(){
   for (i = 0; i < N; i++) {
     a[i] += 1;
     b[i] += 2;
     c[i] += 3;
   }
 }
-
-#pragma omp declare target to(update) device_type(nohost)
+#pragma omp end declare target 
```

Comment:
An `end declare target` cannot occur here.

In OpenMP 5.0, an `end declare target` can only occur after a `declare target` which has no clauses ([spec](https://www.openmp.org/spec-html/5.0/openmpsu62.html)).

This was changed in OpenMP 5.1, where `end declare target` can occur after a `begin declare target` with clauses ([spec](https://www.openmp.org/spec-html/5.1/openmpsu70.html)). My understanding is that without the addition of `begin`, it was difficult to disambiguate the forms with/without the `end`.